osu.Game/OsuGame.cs

```diff
@@ -327,6 +327,8 @@ private void screenChanged(Screen newScreen)
             }
 
             ScreenChanged?.Invoke(newScreen);
+
+            musicController.AllowBeatmapChange = currentScreen.AllowBeatmapChange;

```

Comment:
And here, this could be `musicController.AllowBeatmapChange.BindTo(currentScreen.AllowBeatmapChange)`.